### PR TITLE
newa_traditional first commit

### PR DESCRIPTION
**Pull request by sapradhan**
Hi,
I would like to contribute a new layout for Nepal Bhasa in Newa (prachalit) script which was introduced in Unicode 9.0
Some level of community testing was done via an [online editor](https://sapradhan.keybase.pub/newa-prachalit-trad-ttf/web) based on keyman web.

Thanks

### CODE REVIEW

**Comment by keyman-server** [comment]
This pull request is from an external repo and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

**Comment by LornaSIL** [comment]
Thanks for your first submission!

If you could add the copyright information (use "© 2020 Santosh Pradhan") to these files, that would be great:
* README.md
* LICENSE.md
* newa_traditional.kmn
* newa_traditional.kps (Details tab)

In the README.md, under "Links" please add:
* Home: https://keyman.com/keyboards/newa_traditional

There are various places where you refer to the keyboard name as "newa-traditional". In one place (newa_traditional.php) I saw "Newa Traditional". I prefer "Newa Traditional", but that's up to you. The name should be consistent throughout.

In the .kps file under the "files" tab the graphics from the welcome folder were not included. Those should be added.

Since this is a more rare script and most computers don't have the font installed, I think I'd recommend including font information in the keyboard_info file. You could modify it like this:
```
{
    "license": "mit",
    "languages": { 
         "new-Newae": { 
              "font": { 
              "family": "Noto Sans Newa", 
              "source": "NotoSansNewa-Regular.otf" 
           } 
        }
     },
	  "description": "newa-traditional: Traditional Layout for Nepal Bhasa in Newa (Prachalit) script with dead keys support for special characters"
}
```
Basically, that will help the Web interface show the right font and it will be used on the Android Keyman app.

* On the tablet layout
  * default layer, I think you have the zwj and zwnj switched.
  * shift layer, K_G key - longpress. Please check the character I think the keycap displays U+11442, but it outputs U+11424.
  * shift layer, do you want the word "dead" on the deadkey as you did for the OSK?
* On the phone layout I'm not seeing the zwj/zwnj/dead keys. Should they be there?

Otherwise, I think it's in good shape.

**Comment by sapradhan** [comment]
Thanks for taking time for review, 


> Thanks for your first submission!
> 
> If you could add the copyright information (use "© 2020 Santosh Pradhan") to these files, that would be great:
> 
>     * README.md
> 
>     * LICENSE.md
> 
>     * newa_traditional.kmn
> 
>     * newa_traditional.kps (Details tab)
> 
Done. I have added (cc) Santosh Pradhan to all these files

> In the README.md, under "Links" please add:
> 
>     * Home: https://keyman.com/keyboards/newa_traditional
> 
Added

> 
> There are various places where you refer to the keyboard name as "newa-traditional". In one place (newa_traditional.php) I saw "Newa Traditional". I prefer "Newa Traditional", but that's up to you. The name should be consistent throughout.
> 
Changed all references to Newa Traditional. pls let me know if I missed any.

> In the .kps file under the "files" tab the graphics from the welcome folder were not included. Those should be added.
> 
Added

> Since this is a more rare script and most computers don't have the font installed, I think I'd recommend including font information in the keyboard_info file. You could modify it like this:
> 
> ```
> {
>     "license": "mit",
>     "languages": { 
>          "new-Newae": { 
>               "font": { 
>               "family": "Noto Sans Newa", 
>               "source": "NotoSansNewa-Regular.otf" 
>            } 
>         }
>      },
> 	  "description": "newa-traditional: Traditional Layout for Nepal Bhasa in Newa (Prachalit) script with dead keys support for special characters"
> }
> ```
> 
> Basically, that will help the Web interface show the right font and it will be used on the Android Keyman app.
>
Yes, this would be great. I made the change but after that build.sh validation failed and I have reverted the change. Can you see what is wrong with  https://github.com/keymanapp/keyboards/pull/1084/commits/a04ab3436c62c279e77e585131728b8cbbca9bc9#diff-f4d0a4720bbce2373ef45e9e6d0a387b

>     * On the tablet layout
>       
>       * default layer, I think you have the zwj and zwnj switched.
Thanks for pointing this out. Actually the mistake was in OSK, i have fixed that. 

>       * shift layer, K_G key - longpress. Please check the character I think the keycap displays U+11442, but it outputs U+11424.
I could not find this mistake. Both the keycap and output is U+11424 NEWA LETTER NHA. Can you reverify ? 

>       * shift layer, do you want the word "dead" on the deadkey as you did for the OSK?
Added. However I have questions on dead keys in touch layouts in general.

> 
>     * On the phone layout I'm not seeing the zwj/zwnj/dead keys. Should they be there?
ZWJ/ZWNJ are in K_BACKSLASH (top right key). This is also documented in welcome.html. 

As for dead key, in the desktop layout it was primarily used for symbols. As there is a separate symbols layer in this layout, I did not put it here (in tablet view there was a blank key available so I put it there.). What is the best practice regarding dead keys on touchlayouts, are they necessary as long press menu is available ?

**Comment by darcywong00** [comment]
I wonder if the font should be .ttf instead of .otf

Here's what it looks like when I loaded the keyboard package from this PR
newa

**Comment by sapradhan** [comment]
> I wonder if the font should be .ttf instead of .otf
> 
> Here's what it looks like when I loaded the keyboard package from this PR
> newa

i tried packaging a ttf file but things got worse, even the typed text comes as ? in boxes.
Screenshot_20200301-161710

with .otf, on my android 6 the default layer is rendered with crossed boxes like your screenshot, surprisingly the longpress popup renders characters correctly.

This script is supported with Android 9+, there the keycaps render properly.

**Comment by LornaSIL** [comment]
This rule needs changing from:
```
+ "m"  > U+0220C
```
to 
```
+ "m"  > U+200C
```
The key that seems wrong to me is the TABLET layout, SHIFT layer, K_G longpress. You have U+11442 on the keycap, and the code is U_11424.

Please do make this change again:
```
   "languages": {
        "new-Newa": {
               "font": {
               "family": "Noto Sans Newa",
               "source": "NotoSansNewa-Regular.otf"
           }
    },
```
I think it will help. I notice I gave you the wrong code last time new-Newa :( It could be that was the problem, or more likely it is because I have to trigger the build so it always will fail until I trigger the build.

**Comment by sapradhan** [comment]
> This rule needs changing from:
> 
> ```
> + "m"  > U+0220C
> ```
> 
> to
> 
> ```
> + "m"  > U+200C
> ```
Fixed.


> The key that seems wrong to me is the TABLET layout, SHIFT layer, K_G longpress. You have U+11442 on the keycap, and the code is U_11424.

I still think the keycap is correct, but this happened when I put the cursor **after** the keycap character **for the first time** when you switch to code view.     
keycap-wrong-code

But if you **select** the keycap character it shows the actual character  
keycap-correct-code

It also shows up correctly in design mode
keycap-correct
 
I have also noticed sometimes the Character Identifier remains stuck to the same character until I hide and re-show Character Map. May be its an issue in Keyman Developer ? I am using 13.0.100.0

> Please do make this change again:
> 
> ```
>    "languages": {
>         "new-Newa": {
>                "font": {
>                "family": "Noto Sans Newa",
>                "source": "NotoSansNewa-Regular.otf"
>            }
>     },
> ```
> 
> I think it will help. I notice I gave you the wrong code last time new-Newa :( It could be that was the problem, or more likely it is because I have to trigger the build so it always will fail until I trigger the build.

Committed the changes, however `build.sh` validation fails on my machine 
build-fail

Thank you again for your time and patience

**Comment by LornaSIL** [comment]
The keyboard is now online here: https://keyman.com/keyboards/newa_traditional

**Comment by sapradhan** [comment]
Thanks for accepting. 
However the web version does not render correctly, even when I have Noto Sans Newa locally installed.

image

Should I create a separate issue for this ?

**Comment by darcywong00** [comment]
> Should I create a separate issue for this ?

I've made the issue above
